### PR TITLE
Feature/create menu bar issue#5

### PR DESCRIPTION
**Pull request by suho**
#5 : Custom Navigation Bar
#6 : Add Library 'Page Menu' To Project
#6 : Custom Venue Item Cell
#6 : Connected Side Menu And Page Menu When Add Item Menu

### CODE REVIEW

File: FourSquare/Controllers/Background/BackgroundViewController.swift [hunk 1]
```diff
@@ -12,6 +12,15 @@ import SwiftUtils
 
 class BackgroundViewController: LGSideMenuController {
 
+    // MARK:- Singleton
+
+    class var sharedInstance: BackgroundViewController {
```
Comment (le-binh) [inline FourSquare/Controllers/Background/BackgroundViewController.swift]: Singleton, prefer this way

```
static let sharedInstance = BackgroundViewController()
```

File: FourSquare/Controllers/Base/PageBase/MenuItemViewController.swift [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        let cell = tableView.dequeue(VenueItemTableViewCell)
+        return cell
+    }
+}
+
+//MARK:- Table View Datasource
+
+extension MenuItemViewController: UITableViewDelegate {
+    func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
+        return 140
```
Comment (le-binh) [inline FourSquare/Controllers/Base/PageBase/MenuItemViewController.swift]: Please do not use magic number

File: FourSquare/Controllers/Home/HomeViewController.swift [hunk 3]
```diff
@@ -7,18 +7,124 @@
...
+    var pageMenu: CAPSPageMenu?
+    var itemViewControllers: [UIViewController] = []
+    var activeMenuItems: [ItemMenu] = []
 
-    @IBAction func showSideMenuAction(sender: AnyObject) {
-        UIApplication.sharedApplication().backgroundViewController()?.showHideLeftViewAnimated(true, completionHandler: nil)
+    // MARK:- Life Cycle
+
+    override func viewDidLoad() {
+        self.title = "Explore"
```
Comment (le-binh) [inline FourSquare/Controllers/Home/HomeViewController.swift]: Move string constant to Strings file

File: FourSquare/Controllers/Home/HomeViewController.swift [hunk 4]
```diff
@@ -7,18 +7,124 @@
...
+            }
+            pageViewController.title = defaultItemMenu.title
+            pageViewController.defaultItem = defaultItemMenu
+            viewControllers.append(pageViewController)
+        }
+        return viewControllers
     }
 
+    private func setUpMenuPage(isDefault isDefault: Bool) {
+        var menuItemWidth = (kScreenSize.width - 20) / 3
```
Comment (le-binh) [inline FourSquare/Controllers/Home/HomeViewController.swift]: please do not use magic number

File: FourSquare/Controllers/Home/HomeViewController.swift [hunk 5]
```diff
@@ -7,18 +7,124 @@
...
+        }
+
+    }
+
+    private func setUpNotificationCenter() {
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(self.dismissLeftSideMenu), name: kLGSideMenuControllerWillDismissLeftViewNotification, object: nil)
+    }
+
+    @objc private func dismissLeftSideMenu() {
+        let newActiveMenuItems = BackgroundViewController.sharedInstance.activeMenuItems
```
Comment (le-binh) [inline FourSquare/Controllers/Home/HomeViewController.swift:97]: break down this function into small functions

File: FourSquare/Controllers/Home/HomeViewController.swift [hunk 6]
```diff
@@ -7,18 +7,124 @@
...
+                return
+            }
+            self.setUpMenuPage(isDefault: false)
+        }
+    }
+
+    private func compareTwoArray(oldArray: [ItemMenu], newArray: [ItemMenu]) -> Bool {
+        if oldArray.count == newArray.count {
+            for i in 0..<oldArray.count {
+                if oldArray[i].item.rawValue != newArray[i].item.rawValue {
```
Comment (le-binh) [inline FourSquare/Controllers/Home/HomeViewController.swift]: implement `==` operator for Enum

File: FourSquare/Controllers/Background/BackgroundViewController.swift (hunk 1 above)
Comment (suho) [inline FourSquare/Controllers/Background/BackgroundViewController.swift]: fixed

File: FourSquare/Controllers/Base/PageBase/MenuItemViewController.swift (hunk 2 above)
Comment (suho) [inline FourSquare/Controllers/Base/PageBase/MenuItemViewController.swift]: fixed

File: FourSquare/Controllers/Home/HomeViewController.swift (hunk 3 above)
Comment (suho) [inline FourSquare/Controllers/Home/HomeViewController.swift]: fixed

File: FourSquare/Controllers/Home/HomeViewController.swift (hunk 4 above)
Comment (suho) [inline FourSquare/Controllers/Home/HomeViewController.swift]: fixed

File: FourSquare/Controllers/Home/HomeViewController.swift (hunk 5 above)
Comment (suho) [inline FourSquare/Controllers/Home/HomeViewController.swift:97]: fixed

File: FourSquare/Controllers/Home/HomeViewController.swift (hunk 6 above)
Comment (suho) [inline FourSquare/Controllers/Home/HomeViewController.swift]: fixed

File: FourSquare/Controllers/Home/HomeViewController.swift [hunk 7]
```diff
@@ -7,18 +7,143 @@
...
+
+    }
+
+    private func parametersOfPageMenu(isDefault: Bool) -> [CAPSPageMenuOption] {
+        let menuItemPadding: CGFloat = 5
+        var numberOfItemWillShowOnMenu: CGFloat = 3
+        if !isDefault {
+            numberOfItemWillShowOnMenu = 3.5
+        }
+        let menuItemWidth = (kScreenSize.width - 4 * menuItemPadding) / numberOfItemWillShowOnMenu
```
Comment (le-binh) [inline FourSquare/Controllers/Home/HomeViewController.swift]: let menuItemWidth = isDedault ? kScreenSize.width / 3 : kScreenSize.width / 4

File: FourSquare/Controllers/Home/HomeViewController.swift [hunk 8]
```diff
@@ -7,18 +7,143 @@
...
+        if self.activeMenuItems.count == 0 {
+            self.setUpMenuPage(isDefault: true)
+            return
+        }
+        self.setUpMenuPage(isDefault: false)
+    }
+
+    private func compareTwoArray(oldArray: [ItemMenu], newArray: [ItemMenu]) -> Bool {
+        if oldArray.count == newArray.count {
+            for i in 0..<oldArray.count {
```
Comment (le-binh) [inline FourSquare/Controllers/Home/HomeViewController.swift]: for (index, element) in oldArray.enumerate() {

}

File: FourSquare/Controllers/Background/BackgroundViewController.swift [hunk 9]
```diff
@@ -55,6 +61,7 @@ class BackgroundViewController: LGSideMenuController {
         self.activeMenuItems = self.allMenuItems.filter({
```
Comment (le-binh) [inline FourSquare/Controllers/Background/BackgroundViewController.swift]: self.activeMenuItems = self.allMenuItems.filter({ $0.active })

File: FourSquare/Controllers/Home/HomeViewController.swift (hunk 7 above)
Comment (suho) [inline FourSquare/Controllers/Home/HomeViewController.swift]: fixed

File: FourSquare/Controllers/Home/HomeViewController.swift (hunk 8 above)
Comment (suho) [inline FourSquare/Controllers/Home/HomeViewController.swift]: fixed

File: FourSquare/Controllers/Background/BackgroundViewController.swift (hunk 9 above)
Comment (suho) [inline FourSquare/Controllers/Background/BackgroundViewController.swift]: fixed

**Pull request merged**